rfcs/20240312-standardize-tan-customcall-tuplecollective-ops.md

```diff
@@ -0,0 +1,359 @@
+# [RFC] Standardize TanOp, CustomCallOp with typed FFI, and tuple-collectives ops
+
+Status: Draft<br/>
+Initial version: 03/12/2024<br/>
+Last updated: 03/12/2024<br/>
+Discussion thread: [GitHub](add PR Link)
+
+## Motivation
+
+MHLO `tan` op, `custom_call` op with typed FFI and tuple-collectives ops
```

Comment:
Rewording a bit to be generic "RFC Motivation"

```
Several features have been added to MHLO in the past year, which frameworks want to leverage and members of the community have made requests for them as well. This includes: `TanOp`, `CustomCallOp` with typed FFI (dictionary), and tuple-collective ops (`all_gather`, `all_reduce`, `all_to_all` with variadic operands/results). Some of these features are being used today with various workarounds -- unregistered attributes, serializing dictionary as string, custom_calls. None of these approaches are stable, so we propose adding these ops/features to the StableHLO spec so they can be used safely by the community.
```

Reply:
Thanks, this is better and  appears as a story!